### PR TITLE
SLE-438 replace old APIs calls, remove code branches for old versions

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: org.sonarlint.eclipse.m2e/src/org/sonarlint/eclipse/m2e/internal/M2eUtils.java [hunk 1]
```diff
@@ -65,9 +65,7 @@ private static IFile toSpecificFile(IFile file) {
     IFile finalFile = file;
     IPath rawLocation = file.getRawLocation();
     if (rawLocation != null) {
-      // TODO use getFileForLocation when we will support only Eclipse 4.6+
-      // IFile moreSpecific = ResourcesPlugin.getWorkspace().getRoot().getFileForLocation(rawLocation);
-      IFile moreSpecific = resourceForLocation(rawLocation);
+      IFile moreSpecific = ResourcesPlugin.getWorkspace().getRoot().getFileForLocation(rawLocation);
```
Comment (henryju) [inline org.sonarlint.eclipse.m2e/src/org/sonarlint/eclipse/m2e/internal/M2eUtils.java:68]: Now you should be able to delete method `resourceForLocation` no?

**Review by henryju — CHANGES_REQUESTED** [review]
I think we should add a lower bound on jface depepdency to prevent installation in unsupported versions

File: org.sonarlint.eclipse.m2e/src/org/sonarlint/eclipse/m2e/internal/M2eUtils.java (hunk 1 above)
Comment (kirill-knize-sonarsource) [reply org.sonarlint.eclipse.m2e/src/org/sonarlint/eclipse/m2e/internal/M2eUtils.java:68]: Yes. And method `resourceFor`. Removed both.

Comment (henryju) [reply org.sonarlint.eclipse.m2e/src/org/sonarlint/eclipse/m2e/internal/M2eUtils.java:68]: :+1:

**Review by henryju — APPROVED** [review]

**Pull request merged**